migration/migration.go

```diff
@@ -1,25 +1,130 @@
 package migration
 
 import (
+	"log"
+
 	"github.com/almighty/almighty-core/app"
 	"github.com/almighty/almighty-core/models"
-	"github.com/almighty/almighty-core/remoteworkitem"
+	"github.com/almighty/almighty-core/transaction"
 	"github.com/jinzhu/gorm"
 	"golang.org/x/net/context"
 )
 
-// Perform executes the required migration of the database on startup
-func Perform(ctx context.Context, db *gorm.DB, witr models.WorkItemTypeRepository) error {
-	db.AutoMigrate(
-		&models.WorkItem{},
-		&models.WorkItemType{},
-		&remoteworkitem.Tracker{},
-		&remoteworkitem.TrackerQuery{},
-		&remoteworkitem.TrackerItem{})
-	if db.Error != nil {
-		return db.Error
+// getCurrentVersion returns the highest version from the version
+// table or -1 if that table does not exist.
+//
+// Returning -1 simplifies the logic of the migration process because
+// the next version is always the current version + 1 which results
+// in -1 + 1 = 0 which is exactly what we want as the first version.
+func getCurrentVersion(db *gorm.DB) (int, error) {
+	if !db.HasTable("version") {
+		return -1, nil
+	}
+
+	res := db.Raw("select max(version) from version")
+	if res.Error != nil {
+		return -1, res.Error
 	}
 
+	type VersionResult struct {
+		version int
+	}
+	var v VersionResult
+	res.Scan(&v)
+	return v.version, res.Error
+}
+
+// Func defines the type of function that can be part of a migration sequence
+type Func func(tx *gorm.DB) error
```

Comment:
No need to use an exported name, right? How about renaming it as `migrationFunc` or `function`?